crates/router/src/services/refresh_token.rs

```diff
@@ -0,0 +1,149 @@
+use std::fmt::Debug;
+
+use common_utils::ext_traits::AsyncExt;
+use error_stack::ResultExt;
+
+use crate::{
+    core::{
+        errors::{self, RouterResult},
+        payments,
+    },
+    routes::AppState,
+    services,
+    types::{self, api as api_types, storage},
+};
+
+pub fn connector_supports_access_token(connector: &api_types::ConnectorData) -> bool {
+    match connector.connector_name {
+        api_models::enums::Connector::Globalpay => true,
+        api_models::enums::Connector::Stripe => false,
+        _ => false,
+    }
+}
+
+pub fn router_data_type_conversion<F1, F2, Req1, Req2, Res1, Res2>(
+    router_data: types::RouterData<F1, Req1, Res1>,
+    request: Req2,
+    response: Result<Res2, types::ErrorResponse>,
+) -> types::RouterData<F2, Req2, Res2> {
+    types::RouterData {
+        flow: std::marker::PhantomData,
+        request,
+        response,
+        merchant_id: router_data.merchant_id,
+        address: router_data.address,
+        amount_captured: router_data.amount_captured,
+        auth_type: router_data.auth_type,
+        connector: router_data.connector,
+        connector_auth_type: router_data.connector_auth_type,
+        connector_meta_data: router_data.connector_meta_data,
+        description: router_data.description,
+        router_return_url: router_data.router_return_url,
+        payment_id: router_data.payment_id,
+        payment_method: router_data.payment_method,
+        payment_method_id: router_data.payment_method_id,
+        return_url: router_data.return_url,
+        status: router_data.status,
+        attempt_id: router_data.attempt_id,
+        access_token: router_data.access_token,
+    }
+}
+
+pub async fn add_access_token<
+    F: Clone + 'static,
+    Req: Debug + Clone + 'static,
+    Res: Debug + Clone + 'static,
+>(
+    state: &AppState,
+    connector: &api_types::ConnectorData,
+    merchant_account: &storage::MerchantAccount,
+    router_data: &types::RouterData<F, Req, Res>,
+) -> RouterResult<(
+    Result<Option<types::AccessToken>, types::ErrorResponse>,
+    bool,
+)> {
```

Comment:
It would have been great if you could make this a struct instead. When the return type is a tuple, we have no information on what the bool in the second field signifies.